controller/konnect/konnectextension_controller.go

```diff
@@ -583,12 +575,22 @@ func (r *KonnectExtensionReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 	var updateExtensionStatus bool
 	if ext.Status.Konnect == nil {
+		// Since KonnectControlPlane can be configured in K8s without specifying the cluster type,
+		// this field may be empty and in such a case in Konnect it's set to CLUSTER_TYPE_CONTROL_PLANE
+		// hence assume this as the default value.
+		clusterType := konnectv1alpha1.ClusterTypeControlPlane
+		if cp.Spec.ClusterType != nil {
+			clusterType = konnectClusterTypeToCRDClusterType(
+				sdkkonnectcomp.ControlPlaneClusterType(*cp.Spec.ClusterType),
+			)
+		}
```

Comment:
Perhaps we could refactor that so that all this is handled inside `konnectClusterTypeToCRDClusterType`, WDYT?

Reply:
Ok, so you're leaning towards doing this way, instead of the proposition articulated in _Special notes for your reviewer_? If yes, I agree it makes sense

Reply:
Yup, it does make sense given that an alternative would be to add a net new field to the status which can be extrapolated from the spec field (as it's done via `konnectClusterTypeToCRDClusterType`)

Reply:
OK, so since we're sure that this approach always works. let's do it this way